lib/concurrent/atomic/mutex_count_down_latch.rb

```diff
@@ -9,9 +9,9 @@ class MutexCountDownLatch < Synchronization::LockableObject
 
     # @!macro count_down_latch_method_initialize
     def initialize(count = 1)
-      unless count.is_a?(Integer) && count >= 0
-        raise ArgumentError.new('count must be in integer greater than or equal zero')
-      end
+      Utility::NativeInteger.ensure_integer_and_bounds count
```

Comment:
@pitr-ch this is missing a `require` (context: https://github.com/rack/rack/pull/1198)

Reply:
@matthewd thanks for letting me know.